ocp_tool/co2_interpolation.py

```diff
@@ -0,0 +1,841 @@
+            
+        # Step 3b: Vertical interpolation to match the target grid levels
+        print("Performing vertical interpolation to match target grid levels...")
+        
+        # Start timing for vertical interpolation loop
+        vertical_start_time = time.time()
+        
+        # Define vertical interpolation function for a single target level
+        def interpolate_vertical_level(target_level, icmgg_levels, co2_levels, 
+                                      co2_horizontal_interpolated, level_data, target_shape):
+            # Get the pressure level for the target level
+            target_pressure = None
+            if target_level in level_data and 'pressure' in level_data[target_level]:
+                target_pressure = level_data[target_level]['pressure']
+            
+            # Get array of source CO2 levels (in pressure units) and their corresponding interpolated data
+            src_levels = np.array([float(level) for level in co2_levels])
+            
+            # Initialize the output array
+            interpolated_data = np.zeros(target_shape)
+            
+            # If we have pressure information for the target level, use vertical interpolation
+            if target_pressure is not None:
+                # Reshape for easier indexing
+                rows, cols = target_shape
+                
+                # Process the vertical interpolation for each grid point
+                for i in range(rows):
+                    for j in range(cols):
+                        # Extract CO2 values at this grid point across all source levels
+                        co2_values_at_point = []
+                        valid_levels = []
+                        
+                        for level in co2_levels:
+                            if level in co2_horizontal_interpolated:
+                                value = co2_horizontal_interpolated[level][i, j]
+                                if not np.isnan(value):
+                                    co2_values_at_point.append(value)
+                                    valid_levels.append(float(level))
+                        
+                        # If we have enough valid values, perform 1D interpolation
+                        if len(valid_levels) >= 2:
+                            try:
+                                # Create interpolation function for this grid point
+                                f = interp1d(valid_levels, co2_values_at_point, 
+                                          kind='linear', bounds_error=False, fill_value='extrapolate')
+                                
+                                # Interpolate to the target level
+                                interpolated_data[i, j] = f(target_level)
+                            except Exception:
+                                # Default value if interpolation fails
+                                interpolated_data[i, j] = 0.0006
```

Comment:
Where is this default 0.0006 values is coming from and what units does it have and does it make sense that if interpolation fails that the result shall be a default value and not nan?

Reply:
just for the record, 0.0006 kgCO2 kg-1 converts to ~395 ppm, i.e. approx. from year 2013 AD (so yes, i think setting such a default value is not correct since this would depend on the input file)